### PR TITLE
[libc][math][c23] Add f16fma{,l,f128} C23 math function

### PR DESCRIPTION
**Pull request by overmighty**
Part of #93566.

### CODE REVIEW

File: libc/src/__support/FPUtil/multiply_add.h [hunk 1]
```diff
@@ -39,17 +39,17 @@ multiply_add(T x, T y, T z) {
 #if defined(LIBC_TARGET_CPU_HAS_FMA)
 
 // FMA instructions are available.
-#include "FMA.h"
+#include "src/__support/FPUtil/generic_hardware/fma.h"
```
Comment (overmighty) [inline libc/src/__support/FPUtil/multiply_add.h:42]: I did this to fix a circular dependency where dyadic_float.h was including multiply_add.h -> FMA.h -> generic/FMA.h -> dyadic_float.h

File: libc/config/linux/x86_64/entrypoints.txt [hunk 2]
```diff
@@ -588,6 +589,13 @@ if(LIBC_TYPES_HAS_FLOAT16)
     libc.src.math.ufromfpf16
     libc.src.math.ufromfpxf16
   )
+
+  if(LIBC_TYPES_HAS_FLOAT128)
+    list(APPEND TARGET_LIBM_ENTRYPOINTS
+      # math.h C23 mixed _Float16 and _Float128 entrypoints
+      libc.src.math.f16fma128
```
Comment (lntue) [inline libc/config/linux/x86_64/entrypoints.txt:596]: `f16fmaf128`?

File: libc/config/linux/aarch64/entrypoints.txt [hunk 3]
```diff
@@ -558,6 +560,13 @@ if(LIBC_TYPES_HAS_FLOAT16)
     libc.src.math.ufromfpf16
     libc.src.math.ufromfpxf16
   )
+
+  if(LIBC_TYPES_HAS_FLOAT128)
+    list(APPEND TARGET_LIBM_ENTRYPOINTS
+      # math.h C23 mixed _Float16 and _Float128 entrypoints
+      libc.src.math.f16fma128
```
Comment (lntue) [inline libc/config/linux/aarch64/entrypoints.txt:567]: `f16fmaf128`?

File: libc/src/__support/FPUtil/multiply_add.h (hunk 1 above)
Comment (lntue) [reply libc/src/__support/FPUtil/multiply_add.h:42]: If this is the only place using it right now, maybe we can just directly use `__builtin_fma(f)` here, and adding comments about circular dependency.

**Review by lntue — APPROVED** [review]

**Pull request merged**